src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -143,6 +150,12 @@ public class HeaderDecoder : IRlpValueDecoder<BlockHeader>, IRlpStreamDecoder<Bl
                 blockHeader.BaseFeePerGas = rlpStream.DecodeUInt256();
             }
 
+            if (rlpStream.ReadNumberOfItemsRemaining(null, 1) > 0 &&
+                rlpStream.PeekPrefixAndContentLength().ContentLength == Keccak.Size)
+            {
+                blockHeader.WithdrawalsRoot = rlpStream.DecodeKeccak();
+            }
+
```

Comment:
That is tricky ;) If you receive payloads before the timestamp hash from the CL client. You should be able to decode the payload and later return to CL that the payload is invalid because it has withdrawals. There are hive tests that are checking it.